### DISCUSSION TITLE
Cannot run commands to make sherlock work

### DISCUSSION THREAD

**Discussion by Joe-Swanson000** [sherlock-project/sherlock#1357] [General]
## Checklist

- [x] I'm reporting a bug in Sherlock's functionality
- [ ] The bug I'm reporting is not a false positive or a false negative
- [x] I've verified that I'm running the latest version of Sherlock
- [x] I've checked for similar bug reports including closed ones
- [ ] I've checked for pull requests that attempt to fix this bug

## Description

WRITE DESCRIPTION HERE

I have just recently installed sherlock, and I cannot get it to run properly. I have installed the latest version, and also have Python and Pip. Here is how I went about my installation process:
1.) Installed Sherlock Zip folder on desktop
2.) Extracted all and made a new folder on desktop
3.) Installed Python 3.10.2 windows installer and have completed the install process.
4.) Typed in this command into cmd running as admin: cd C:\Users\19706\Desktop\sherlock-master
5.) Ran the command: pip3 install –r requirements.txt. Then get and error that says; "ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements.txt'

Please help with this. I've been trying to find videos and articles on this, but there has been 0 help as of now.

I have also tried to clone the repo, but I get this message: git' is not recognized as an internal or external command, operable program or batch file.

![image](https://user-images.githubusercontent.com/98627814/151645882-0ea237bb-184d-4f47-b664-d93e6f2ce3bc.png)

![image](https://user-images.githubusercontent.com/98627814/151645962-e6ddaf99-d035-4651-915d-a8d5df9dd058.png)

![image](https://user-images.githubusercontent.com/98627814/151645854-8fe86de8-ed9b-4802-a4ff-278a13b1b778.png)

![image](https://user-images.githubusercontent.com/98627814/151646313-0a6652ba-ba18-4336-bc15-9fe5dfbf4dfc.png)

![image](https://user-images.githubusercontent.com/98627814/151646350-cd2b892d-091b-499a-b28b-771207b76d63.png)

![image](https://user-images.githubusercontent.com/98627814/151646504-a278edb4-515f-4a1f-936e-3c062eca29cc.png)

**Comment by Ishanned** [comment]
Regarding git command :
It seems git is not installed properly. Try running `git --version` in the command prompt. If the same issue shows up, then try to uninstall git and install afresh. 
Also can I have a look inside sherlock-master folder of yours?

**Comment by Joe-Swanson000** [comment]
> Regarding git command : It seems git is not installed properly. Try running `git --version` in the command prompt. If the same issue shows up, then try to uninstall git and install afresh. Also can I have a look inside sherlock-master folder of yours?

Here is what is inside of my sherlock master folder. Also, can you link to me afresh? I tried to do git --version, but it just led me to this: 
![image](https://user-images.githubusercontent.com/98627814/151687164-e74e719a-fe1e-4d6d-9468-e30bf6cac6f5.png)


![image](https://user-images.githubusercontent.com/98627814/151687107-c558fff6-7d06-40c9-8708-3415a2fe6e92.png)

![image](https://user-images.githubusercontent.com/98627814/151687114-2ba0e7f1-f99f-478a-a1b1-3cbb8d406058.png)

![image](https://user-images.githubusercontent.com/98627814/151687126-152dd8d0-a45e-4f76-bb5f-339987d2979d.png)

![image](https://user-images.githubusercontent.com/98627814/151687134-a54ef39d-4afe-44e8-9705-3ead3314def0.png)

![image](https://user-images.githubusercontent.com/98627814/151687142-cdfde397-af63-4847-af8d-2ca88441b895.png)

**Comment by sivaprassath99** [comment]
Try to add `PATH` to the GIT.
If your on Windows
Try this,
- Go to My Computer => Local Disk(C:) => Program Files(x86) => Git => cmd
- Right Click the git => Select Properties
- Under the location Copy the text eg - C:\Program Files (x86)\Git\cmd
- Come back to the Desktop
- Right-click My Computer
   - Select property
   - Open Advanced and Click Environment Variables
   - In the System variables Find the Variable call Path
   - Click the variable
   - Click the Edit Button
   - Select the Variable value Text Box .
   - Add a semicolon (;) in last
   - Then Right-click and press Paste
   - Press Apply and OK

**Comment by Joe-Swanson000** [comment]
> 

I don't have git though, that's the thing. And I've tried to reinstall python and sherlock, but I still don't have Git. Is it like a separate program I have to install?

**Comment by sivaprassath99** [comment]
>I don't have git though, that's the thing. And I've tried to reinstall python and sherlock, but I still don't have Git. Is it like a separate program I have to install?

Yes, You have to install Git separately.

**Comment by Ishanned** [comment]
If you want to get started right away, one thing you can do is open requirements.txt, then go to command prompt, type `pip install variable` , where you replace `variable` with each item in the list.
Like for the first element, you type in `pip install certifi>=2019.6.16`, then let it install and then proceed with other items in the file.
Note: as a programmer, its a bit unhealthy practice as this does not follow DRY.

**Comment by JDR11** [comment]
@Joe-Swanson000 I believe it is as simple as you were not running the commands from the correct folder. 
You were running it from:
C:\Users\19706\Desktop\sherlock-master
But with the way you have unzipped it you should be running the command from:
C:\Users\19706\Desktop\sherlock-master\sherlock-master

**Comment by Joe-Swanson000** [comment]
> > I don't have git though, that's the thing. And I've tried to reinstall python and sherlock, but I still don't have Git. Is it like a separate program I have to install?
> 
> Yes, You have to install Git separately.

- Thank you. I installed Git and ran the commands through there, but I did get an error message. I did the command: 
"$ git clone https://github.com/sherlock-project/sherlock.git"
- And had it downloaded. Then I cd'd into sherlock: "cd sherlock"
- Lastly I ran "pip3 install -r requirements.txt" and downloaded everything. I did though get something near the end that read:

"You should consider upgrading via the 'C:\Users\19706\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\python.exe -m pip install --upgrade pip' command."

Don't know if I should get it, but nonetheless, I digress. This next step is where I encountered the problems. The command: "python3 sherlock.py Marss_NE" Marss has a Twitter acct, so I wanted to see if it would work, but all I got was the error: 

"C:\Users\19706\AppData\Local\Microsoft\WindowsApps\python3.exe: can't open file 'C:\\Users\\19706\\sherlock\\sherlock.py': [Errno 2] No such file or directory"

![image](https://user-images.githubusercontent.com/98627814/151890226-ed448ebd-d90a-49ab-8a49-70babc6cedb0.png)

**Comment by Joe-Swanson000** [comment]
> If you want to get started right away, one thing you can do is open requirements.txt, then go to command prompt, type `pip install variable` , where you replace `variable` with each item in the list. Like for the first element, you type in `pip install certifi>=2019.6.16`, then let it install and then proceed with other items in the file. Note: as a programmer, its a bit unhealthy practice as this does not follow DRY.

I decided to download git, and got most of the commands working aside from 1. Thank you for the help though

**Comment by Joe-Swanson000** [comment]
> @Joe-Swanson000 I believe it is as simple as you were not running the commands from the correct folder. You were running it from: C:\Users\19706\Desktop\sherlock-master But with the way you have unzipped it you should be running the command from: C:\Users\19706\Desktop\sherlock-master\sherlock-master

This makes sense actually. If installing git didn't work, then I'd probably have done this.

**Comment by sivaprassath99** [comment]
>Don't know if I should get it, but nonetheless, I digress. This next step is where I encountered the problems. The command: "python3 sherlock.py Marss_NE" Marss has a Twitter acct, so I wanted to see if it would work, but all I got was the error:

>"C:\Users\19706\AppData\Local\Microsoft\WindowsApps\python3.exe: can't open file 'C:\Users\19706\sherlock\sherlock.py': [Errno 2] No such file or directory"

Check whether the file is that directory. You may have been trying this in another directory.

**Comment by Joe-Swanson000** [comment]
> > Don't know if I should get it, but nonetheless, I digress. This next step is where I encountered the problems. The command: "python3 sherlock.py Marss_NE" Marss has a Twitter acct, so I wanted to see if it would work, but all I got was the error:
> 
> > "C:\Users\19706\AppData\Local\Microsoft\WindowsApps\python3.exe: can't open file 'C:\Users\19706\sherlock\sherlock.py': [Errno 2] No such file or directory"
> 
> Check whether the file is that directory. You may have been trying this in another directory.

The thing is, I do have sherlock.py, and python

![image](https://user-images.githubusercontent.com/98627814/152095793-a1e892f4-722b-43cd-81fe-d36f25fa3f7d.png)

**Comment by JDR11** [comment]
You mentioned earlier there was one command which wasn't working.
Which one was it? Was it perhaps the `pip3 install -r requirements.txt`?
Edit: Ignore please, didn't read previous comments thoroughly.
